### PR TITLE
DOC: Add development installation instructions

### PR DESCRIPTION
**Pull request by tswast**
Includes conda install steps for dependencies.

Closes https://github.com/pydata/pandas-gbq/issues/63.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+
+.. code-block:: shell
+
+    $ conda develop .
+
+Install the necessary dependencies
+
+.. code-block:: shell
+
+    $ conda install --channel conda-forge \
```
Comment (jreback) [inline docs/source/install.rst]: so the point of this is you *don't* have to specify secondary deps, they will simply be installed via the dependency graph. it looks like you install

``conda create -n foo --dry google-auth-oauthlib google-api-python-client pandas google-auth-httplib2``

though I am suprised that ``google-auth`` is actually a dep of the ``google-auth-oauthlib`` (and httplib2), you usually don't do it this way, rather installing ``google-auth`` should install all libs under its tree, so this is a bit odd.

File: docs/source/install.rst [hunk 2]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+.. code-block:: shell
+
+   $ conda create --no-deps -n pandas-gbq
+   $ source activate pandas-gbq
+
+Install pandas-gbq in development mode
+
+.. code-block:: shell
+
+    $ conda develop .
```
Comment (jreback) [inline docs/source/install.rst]: we use ``python setup.py develop`` rather than ``conda develop``

File: docs/source/install.rst [hunk 3]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+
+.. code-block:: shell
+
+    $ conda install conda-build
+
+Optionally, create a new conda environment.
+
+.. code-block:: shell
+
+   $ conda create --no-deps -n pandas-gbq
```
Comment (jreback) [inline docs/source/install.rst]: you can do this all in one go, see below

File: docs/source/install.rst [hunk 4]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+
+Conda
+~~~~~
+
+To install packages in development mode using conda, first
+install the conda build tools package in your root conda environment.
+
+.. code-block:: shell
+
+    $ conda install conda-build
```
Comment (jreback) [inline docs/source/install.rst]: you don't need conda-build

File: docs/source/install.rst [hunk 5]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+
+Optionally, create a new conda environment.
```
Comment (jreback) [inline docs/source/install.rst]: this is not optional, you *always* do this.

File: docs/source/install.rst [hunk 6]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+    $ cd pandas-gbq
+
+Pip
+~~~
+
+To install pandas-gbq and its dependencies in development mode via pip
+
+.. code-block:: shell
+
+    $ pip install -e .
```
Comment (jreback) [inline docs/source/install.rst]: we generally don't use pip to install things when we have proper conda packages. sure you can, but its not the first thing you do.

Comment (tswast) [inline docs/source/install.rst]: You're right.

I'd like to keep these instructions for folks that aren't using conda. I've moved the section after and added a disclaimer that it's not necessary if using conda.

File: docs/source/install.rst [hunk 7]
```diff
@@ -32,6 +32,60 @@ Install from Source
...
+
+Conda
+~~~~~
+
+To install packages in development mode using conda, first
+install the conda build tools package in your root conda environment.
```
Comment (tswast) [inline docs/source/install.rst]: I've moved these instructions all to contributing.rst.

File: docs/source/install.rst (hunk 4 above)
Comment (tswast) [inline docs/source/install.rst]: removed.

File: docs/source/install.rst (hunk 3 above)
Comment (tswast) [inline docs/source/install.rst]: Done.

**Comment by jreback** [comment]
yeah see #20 about the coverage issue. Maybe need to skip the user checks for this.

**Pull request merged**